rust/arrow/src/compute/kernels/cast.rs

```diff
@@ -203,6 +204,19 @@ pub fn cast(array: &ArrayRef, to_type: &DataType) -> Result<ArrayRef> {
             Int64 => cast_numeric_to_string::<Int64Type>(array),
             Float32 => cast_numeric_to_string::<Float32Type>(array),
             Float64 => cast_numeric_to_string::<Float64Type>(array),
+            Binary => {
+                let from = array.as_any().downcast_ref::<BinaryArray>().unwrap();
+                let mut b = StringBuilder::new(array.len());
+                for i in 0..array.len() {
+                    if array.is_null(i) {
+                        b.append(false)?;
+                    } else {
+                        b.append_value(str::from_utf8(from.value(i)).unwrap())?;
```

Comment:
`from_utf8` can panic if the data is not valid utf8 bytes. Perhaps it's better to convert failures to null values instead? This would behave similarly to other casts that return nulls on overflowing data.

Reply:
Why we can't return error? I don't it's correct to return null rather than error.

Reply:
@liurenjie1024 +1

Reply:
We've had the broader discussion around casts, on whether to return an error on overflows/invalid data, or whether to expose an option to the user to define their desired behaviour. I haven't opened a JIRA for this, perhaps I should, so we can address the overall cast behaviour there?

For now, an expedient would actually not be to downcast to `BinaryArray`, but to create a `StringArray` from the binary data, but maybe that's a premature optimisation.

Reply:
Another discussion about configuring behaviors of error handling would be fine. But I can't understand for now why we can't just return error? Inconsistency with existing behavior?

Reply:
Yes, inconsistency. For example, if you cast a `i64` to `u8`, negative values and values that don't fit into `u8` will be cast to nulls.

Reply:
Here's the CPP equivalent: https://github.com/apache/arrow/blob/master/cpp/src/arrow/compute/kernels/cast.h#L38

Reply:
For this PR, I agree that we can return null to match existing behavior. But I don't think it's good idea to make returning null as default behavior, but returning error should be. Please open an jira ticket to track this.

Reply:
Done: https://issues.apache.org/jira/browse/ARROW-7364